### PR TITLE
feat(cors-proxy): add and collect http metrics

### PR DESCRIPTION
**Pull request by IvoGoman**
## Description

Added metrics:
 - "http_requests_total"
 - "request_duration_seconds"
 - "response_size_bytes"

 Metrics are exposed by default on 6543
 ServiceMonitor added to collect the metrics

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Related Issue #286

> Remove if not applicable

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [ ] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by kengou — APPROVED** [review]
lgtm

File: cmd/cors-proxy/instrumentation.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+			Help:    "A histogram of response sizes for requests.",
+			Buckets: []float64{0, 512, 4096, 16384, 65536, 262144, 1048576, 4194304, 16777216},
+		},
+		[]string{"code", "method"},
+	)
+
+	// Register all of the metrics in the standard registry.
+	registry.MustRegister(requestsTotal, requestDuration, responseSize)
+
+	intrumentedHandler := promhttp.InstrumentHandlerCounter(
```
Comment (uwe-mayer) [inline cmd/cors-proxy/instrumentation.go:43]:
```suggestion
	intrumentedHandler := promhttp.InstrumentHandlerCounter(
```

**Review by uwe-mayer — APPROVED** [review]
LGTM, two comments

File: cmd/cors-proxy/main.go [hunk 2]
```diff
@@ -69,14 +75,15 @@ func init() {
 
 	pflag.StringVar(&host, "host", envHost, "")
 	pflag.Uint64Var(&port, "port", portParsed, "")
+	pflag.Uint64Var(&metricsPort, "metrics-addr", 6543, "port for metrics")
```
[the remark this answers is not in the document]
Comment (IvoGoman) [reply cmd/cors-proxy/main.go]: Absolutely  👍

**Pull request merged**